src/primitives.cc

```diff
@@ -507,11 +502,11 @@ const Geometry *PrimitiveNode::createGeometry() const
 				z2 = this->h;
 			}
 
-			auto circle1 = new point2d[fragments];
-			auto circle2 = new point2d[fragments];
+			auto circle1 = std::vector<point2d>(fragments);
+			auto circle2 = std::vector<point2d>(fragments);
 
-			generate_circle(circle1, r1, fragments);
-			generate_circle(circle2, r2, fragments);
+			generate_circle(circle1.data(), r1, fragments);
+			generate_circle(circle2.data(), r2, fragments);
```

Comment:
To further improve this, do you think it would be better and/or safer to pass an iterator (e.g. from the std::vector back_inserter()) instead of passing a raw pointer?

Reply:
That depends on if you wanted to use `generate_circle` with other containers. I would personally keep it as-is, and would only change it later if necessary.

Reply:
an iterator would be container agnostic, but we'd have to choose an iterator. I'll keep it like this